# Viral Ecology of Avian Influenza Viruses in Waterfowl of the Central Valley of California using MinION Sequencing

## Script: analysis_notebook.sh
This R mardown notebook provides an interactive way to run all the commands used to generate the results provided in the manuscript. It is customizable and interactive.

This script is part of the following manuscript:
### "Prevalence and Viral Ecology of Avian Influenza of Waterfowl in the Central Valley of California"  
###Madeline M. McCuen | Maurice E. Pitesky | Jeff J. Buler | Sarai Acosta | Alexander Wilcox | Ronald F. Bond | Samuel L. Díaz-Muñoz

First, please review the README file in the repository to configure your environment properly. 
Ok, on with the script :) 

# 1. Sequecing Data Description and Processing

## 1.1 Data Description
One hundred seventy-five swab samples were collected from live-trapped or hunter-killed waterfowl in two wetland basins in the Central Valley of California: the Delevan National Wildlife Refuge and the Sacramento National Wildlife Refuge Complex. 

RNA was extracted from these swab samples (per bird: one pharyngeal and one cloacal, which were stored in the same tube, thus pooled) and amplified using universal Influenza A primers (Zhou et al 2009), but modified with tails to use the Oxford Nanopore PCR barcoding kit. Resulting amplicons were sequenced in two separate, new MinION flow cells (R.10.4) using Q20+ chemistry.

## 1.2 Basecalling and Demultiplexing
This step requires access to the raw fast5 files and a high performance GPU. The repository mostly starts analysis after files have been basecalled and demultiplexed. 

First generate statistics on raw fast5 files

```{bash}
#Count fast5 reads
```

Second, basecall and demultiplex reads

```{bash}
#Make sure to run recursive, including all folders in each run
#Make sure to include read-splitting in the basecalling or demultiplexing command
#Remember have two runs that share barcodes, so have to process separately and append run name to the front
#Include calibration: ask Ivy if she included in run

#Run 1 Basecalling 
~/ont-guppy/bin/guppy_basecaller --input_path /mnt/data0/MinION_reads/pitesky_avian_samples1/pitesky_avian_samples1 --save_path /mnt/data0/sam/pitesky_avian_samples1/fastq_hac --config dna_r10.4_e8.1_hac.cfg -x cuda:all:100% --num_callers 256 --recursive --calib_detect
#Looks like read-splitting in basecalling not avail on Guppy 5.11?

#Start here on demultiplexing Oct 28, 2022

#Run 1 Demultiplexing
~/ont-guppy/bin/guppy_barcoder --require_barcodes_both_ends -i /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/20211216_1317_MN23913_FAR27359_f42a3dea/fastq_sac_cd/split_pass -s /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/20211216_1317_MN23913_FAR27359_f42a3dea/fastq_sac_cd/demultiplexed_split_pass --arrangements_files "barcode_arrs_pcr12.cfg barcode_arrs_pcr96.cfg" -x cuda:0 --worker_threads 8

#Run 2 Basecalling
~/ont-guppy/bin/guppy_basecaller --input_path /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/no_sample/20211217_1150_MN23913_FAR27359_a1bdbce7/fast5_pass/ --save_path /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/no_sample/20211217_1150_MN23913_FAR27359_a1bdbce7/fastq_hac_cd --calib_detect --config dna_r10.4_e8.1_hac.cfg -x cuda:0 --num_callers 8

#Run 2 Demultiplexing
~/ont-guppy/bin/guppy_barcoder --require_barcodes_both_ends -i /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/20211216_1317_MN23913_FAR27359_f42a3dea/fastq_sac_cd/split_pass -s /mnt/data0/MinION_reads/pitesky_avian_samples1_cp/pitesky_avian_samples1/20211216_1317_MN23913_FAR27359_f42a3dea/fastq_sac_cd/demultiplexed_split_pass --arrangements_files "barcode_arrs_pcr12.cfg barcode_arrs_pcr96.cfg" -x cuda:0 --worker_threads 8
```

## 1.3 Basecalling and Demultiplexing Statistics

```{bash}

```

# 2. Generate Complete Genomes from Swab Samples
Going to start with some dummy samples from:
`pitesky_avian_samples1_cp_bc/pitesky_avian_samples1/20211216_1317_MN23913_FAR27359_f42a3dea/fastq_hac_cd/demultiplexed_split_pass/`

I picked barcodes 73-76, which I will arbitrarily put into Run A = 73&74 and Run B = 75-76

First Task is to trim adapters and keep reads we want

```{bash}
#Let's start with one barcode and then write script to scale up to others
srun -t12:00:00 -c6 --mem=30000 --threads=6 --pty /bin/bash
source activate gbbseq-env
module load usearch/8.1.1861

#First let's concatenate the FASTQs in the directory into one
cat data/runA/barcode73/*.fastq > barcode73.fastq

#Check number of reads
wc -l barcode73.fastq | awk '{print $1/4}'
#98,767
#98k+ reads!

cutadapt -a TTTCTGTTGGTGCTGATATTG...GAAGATAGAGCGACAGGCAAGT -a ACTTGCCTGTCGCTCTATCTTC...CAATATCAGCACCAACAGAAA -o ont_landing_pad_barcode73.fastq barcode73.fastq --action=lowercase
#Quick check with less, suggests that at least in some reads 

cutadapt -a AGCAAAAGCAGG...CCTTGTTTCTACT -a AGTAGAAACAAGG...CCTGCTTTTGCT -o unitrimmed_barcode73.fastq ont_landing_pad_barcode73.fastq --discard-untrimmed

#Total reads processed:                  98,767
#Reads with adapters:                    96,619 (97.8%)

#== Read fate breakdown ==
#Reads discarded as untrimmed:            2,148 (2.2%)
#Reads written (passing filters):        96,619 (97.8%)

#Total basepairs processed:    64,587,467 bp
#Total written (filtered):     42,200,543 bp (65.3%)

#Looks like a lot of merged reads

```


# 3. Match Sequences to Flu DB 

# 4. AIV Sequencing Statistics
Per read
Complete genomes per sample

# 4. Avian Ecology 

## 4.1 Prevalence per location

## 4.2 AIV Subtype Information
General, per location,

## 4.3 AIV Host Information
Predicted by sequence (but multiple?) versus actual

## 4.4. Comparison to CWT abundance data? Waterfowl Density versus Positives

## 4.5. Phylogeographic Analysis (maybe...)
Clades and diversity by geographic location
-Do subtypes cluster by pond/location?